pkg/discovery/service.go

```diff
@@ -39,16 +41,17 @@ func NewDiscovery(ol *overlay.Cache, kad *kademlia.Kademlia, stat statdb.DB) *Di
 // We currently do not penalize nodes that are unresponsive,
 // but should in the future.
 func (d *Discovery) Refresh(ctx context.Context) error {
-	// TODO(coyle): make refresh work by looking on the network for new ndoes
+	var errs []error
 	nodes := d.kad.Seen()
-
-	for _, v := range nodes {
-		if err := d.cache.Put(ctx, v.Id, *v); err != nil {
-			return err
+	for _, node := range nodes {
+		_, err := d.kad.Ping(ctx, *node)
+		if err != nil {
+			zap.L().Debug("error pinging node - discovery.Refresh", zap.Error(err))
```

Comment:
Should Discovery have its own logger instead?

Reply:
Discovery probably should have it's own logger, you're right